20_beta_diversity.Rmd

```diff
@@ -247,6 +247,151 @@ As a final note, `mia` provides functions for the evaluation of additional dissi
+                               p_value = anova.cca(rda)["Model", "Pr(>F)"] )
+
+# Loop through variables
+permutations <- 999
+for( variable_name in variable_names ){
+    # Create a formula
+    formula <- as.formula(paste0("assay ~ ", variable_name) )
+    # Perform RDA
+    rda_temp <- rda(formula, data = coldata, scale = TRUE, na.action = na.exclude)
+    # Add Info to list
+    rda_info[[variable_name]] <- c(constrained = rda_temp$CCA$tot.chi, 
+                                   unconstrainded = rda_temp$CA$tot.chi, 
+                                   proportion = rda_temp$CCA$tot.chi/rda$CA$tot.chi, 
+                                   p_value = anova.cca(rda_temp, permutations = permutations
+                                                       )["Model", "Pr(>F)"] )
+}  
+# Convert into data.frame
+rda_info <- t(as.data.frame(rda_info))
+rda_info_clean <- rda_info
+# Adjust names
+colnames(rda_info_clean) <- 
+    c("Explained by variables", "Unexplained by variables", "Proportion expl by vars", 
+      paste0("P-value (PERMANOVA ", permutations, " permutations)") )
+# Print info
+kable(rda_info_clean)
+```
+
+```{r microbiome_RDA2}
+# Load ggord for plotting
+if(!require("ggord")){
+    if(!require("devtools")){
+        install.packages("devtools")
+        library("devtools")
+    }
+    install_github("https://github.com/fawda123/ggord/")
+    library("ggord")
+}
+if(!require("ggplot2")){
+    install.packages("ggplot2")
+    library("ggplot2")
+}
+# Since na.exclude was used, if there were rows missing information, they were 
+# dropped off. Subset coldata so that it matches with rda.
+coldata <- coldata[ rownames(rda$CCA$wa), ]
+
+# Adjust names
+# Get labels of vectors
+vec_lab_old <- rownames(rda$CCA$biplot)
+
+# Loop through vector labels
+vec_lab <- sapply(vec_lab_old, FUN = function(name){
+    # Get the variable name
+    variable_name <- variable_names[ str_detect(name, variable_names) ]
+    # If the vector label includes also group name
+    if( !any(name %in% variable_names) ){
+        # Get the group names
+        group_name <- unique( coldata[[variable_name]] )[ 
+        which( paste0(variable_name, unique( coldata[[variable_name]] )) == name ) ]
+        # Modify vector so that group is separated from variable name
+        new_name <- paste0(variable_name, " \U2012 ", group_name)
+    } else{
+        new_name <- name
+    }
+    # Add percentage how much this variable explains, and p-value
+    new_name <- expr(paste(!!new_name, " (", 
+                           !!format(round( rda_info[variable_name, "proportion"]*100, 1), nsmall = 1), 
+                           "%, ",italic("P"), " = ", 
+                           !!gsub("0\\.","\\.", format(round( rda_info[variable_name, "p_value"], 3), 
+                                                       nsmall = 3)), ")"))
+
+    return(new_name)
+})
+# Add names
+names(vec_lab) <- vec_lab_old
```

Comment:
Is there a chance to simplify the example? Ideally, the OMA examples would avoid giving too much technical details that are unrelated to the main topic (here RDA). This would be pedagogically better.

Reply:
For figure tuning, one could then cite the visualization chapter, for instance, and more tricks can be there.

Reply:
Yep, it is possible, I just copy-pasted my code

Reply:
Great. With OMA, also more generally, we really need to pay attention to readability by non-experts.